### PR TITLE
APIClient ws client needs to be usable from browser - Closes #6036

### LINKED ISSUE #6036: WSClient from lisk-api-client needs to be usable from browser
### Expected behavior
ws client should be usable in browser environment

### Actual behavior
```
Uncaught (in promise) TypeError: _a.on is not a function
    at ws_channel.ts:74
    at new Promise (<anonymous>)
    at WSChannel.connect (ws_channel.ts:73)
    at Object.push../node_modules/@liskhq/lisk-api-client/dist-node/create_clients.js.exports.createWSClient (create_clients.ts:39)
```

### Additional Information
- ws_channel in api-client is using non-isomorphic functions. It should use `onmessage`, `onopen` and `onerror`
-  `Error: Request is not acknowledged in 2000m` shows up on browser with below step (without `invoke`)

### Steps to reproduce
open `index.js` in `elements/lisk-client` and run
```
(async () => { const client =lisk.apiClient.createWSClient('ws://localhost:8080/ws'); })()
```

### Which version(s) does this affect? (Environment, OS, etc...)
5.0.0-

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

This PR resolves #6036

### How was it solved?

- Used isomorphic interfaces for websocket 
- Added e2e tests to verify 

### How was it tested?

- Run all tests

### CODE REVIEW

File: elements/lisk-api-client/src/ws_channel.ts [hunk 1]
```diff
@@ -108,15 +111,33 @@ export class WSChannel {
...
-				this._ws?.terminate();
-				this.isAlive = false;
-				this._ws = undefined;
-				resolve();
-			});
-			this._ws?.close();
+		if (this._ws.readyState === WebSocket.CLOSED) {
+			this.isAlive = false;
+			this._ws = undefined;
+			return Promise.resolve();
```
Comment (shuse2) [inline elements/lisk-api-client/src/ws_channel.ts:117]:
```
return;
```
should be ok here

File: elements/lisk-api-client/src/ws_channel.ts [hunk 2]
```diff
@@ -108,15 +111,33 @@ export class WSChannel {
...
+			retry();
 		});
+
+		this._ws.close();
+		await Promise.race([
+			disconnect,
+			timeout(CONNECTION_TIMEOUT, `Could not disconnect in ${CONNECTION_TIMEOUT}ms`),
+		]);
+
+		return Promise.resolve();
```
Comment (shuse2) [inline elements/lisk-api-client/src/ws_channel.ts:140]: same as above

File: elements/lisk-api-client/src/ws_channel.ts [hunk 3]
```diff
@@ -131,6 +152,12 @@ export class WSChannel {
 		};
 
 		const send = new Promise((resolve, reject) => {
+			if (!isNode) {
+				this._ws?.send(JSON.stringify(request));
+				resolve();
+				return;
+			}
+
```
Comment (shuse2) [inline elements/lisk-api-client/src/ws_channel.ts:160]: in invoke, the send is not meant to be acknowledge timeout, the `Response not received in X` should be the acknowledge timeout.
ie) send it self doesn't need the promise

File: elements/lisk-api-client/src/ws_channel.ts [hunk 4]
```diff
@@ -108,15 +111,33 @@ export class WSChannel {
...
-				resolve();
-			});
-			this._ws?.close();
+		if (this._ws.readyState === WebSocket.CLOSED) {
+			this.isAlive = false;
+			this._ws = undefined;
+			return Promise.resolve();
+		}
+
+		const disconnect = new Promise<void>(resolve => {
```
Comment (shuse2) [inline elements/lisk-api-client/src/ws_channel.ts:120]: I think no need to retry the disconnect

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: elements/lisk-api-client/src/ws_channel.ts (hunk 1 above)
Comment (nazarhussain) [reply elements/lisk-api-client/src/ws_channel.ts:117]: Yes it should be but ESLint is complaining for consistent promise return.

File: elements/lisk-api-client/src/ws_channel.ts (hunk 2 above)
Comment (nazarhussain) [reply elements/lisk-api-client/src/ws_channel.ts:140]: Same here its some ESLint issue that it complaints about consistent promise return.

File: elements/lisk-api-client/src/ws_channel.ts (hunk 3 above)
Comment (nazarhussain) [reply elements/lisk-api-client/src/ws_channel.ts:160]: Done

File: elements/lisk-api-client/src/ws_channel.ts (hunk 4 above)
Comment (nazarhussain) [reply elements/lisk-api-client/src/ws_channel.ts:120]: Done

File: elements/lisk-api-client/src/ws_channel.ts (hunk 1 above)
Comment (shuse2) [reply elements/lisk-api-client/src/ws_channel.ts:121]: if you consistently use `return`, it should be ok, also the bottom return is not necessary

File: elements/lisk-api-client/src/ws_channel.ts [hunk 5]
```diff
@@ -130,14 +149,9 @@ export class WSChannel {
...
-			this._ws?.send(JSON.stringify(request), (err): void => {
-				if (err) {
-					return reject(err);
-				}
-
-				return resolve();
-			});
+		const send = new Promise(resolve => {
+			this._ws?.send(JSON.stringify(request));
+			resolve();
```
Comment (shuse2) [inline elements/lisk-api-client/src/ws_channel.ts:154]: not strictly related to this PR, but the send doesn't need to be wrapped with promise.
`ACKNOWLEDGMENT_TIMEOUT` should be used in place of `RESPONSE_TIMEOUT`

File: elements/lisk-api-client/src/ws_channel.ts [hunk 6]
```diff
@@ -70,53 +70,72 @@ export class WSChannel {
 
 	public async connect(): Promise<void> {
 		this._ws = new WebSocket(this._url);
+		// this._ws.onclose = this._handleClose.bind(this);
```
Comment (shuse2) [inline elements/lisk-api-client/src/ws_channel.ts:73]: why commented out here?

**Review by shuse2 — CHANGES_REQUESTED** [review]
LGTM in general!

**Review by shuse2 — APPROVED** [review]

**Review by Incede — APPROVED** [review]

**Pull request merged**